src/snowflake/snowpark/dataframe_na_functions.py

```diff
@@ -63,6 +63,16 @@ def _is_value_type_matching_for_na_function(
     )
 
 
+def _check_subset_parameter(subset: Optional[Union[str, Iterable[str]]]) -> None:
+    """Produces exception when invalid subset parameter was passed."""
+    if (
+        subset is not None
+        and not isinstance(subset, str)
+        and not isinstance(subset, (list, tuple))
+    ):
+        raise TypeError("subset should be a list or tuple of column names")
```

Comment:
```suggestion
        raise TypeError("subset should be a single column name, list or tuple of column names")
```

Reply:
I like the suggestion, sounds much better. Will have to update existing tests.